Hannah-Sten/TeXiFy-IDEA
# IntelliJ built-in pdf viewer - the beginning

Fix #1625

#### Summary of additions and changes

* Adds a pdf viewer that uses the [IntelliJ PDF Viewer](https://github.com/FirstTimeInForever/intellij-pdf-viewer) to show the pdf in IntelliJ. Right now, it only displays the pdf after compilation.  
* This viewer is only available when the IntelliJ PDF Viewer plugin is installed and enabled. We ensure this by providing a custom pdf viewer extension point and registering this extension point from the optional plugin dependency xml of the IntelliJ PDF Viewer plugin.

Some known bugs/future improvements:
* After a reload it automatically shows the first page on default zoom. ([#11](https://github.com/FirstTimeInForever/intellij-pdf-viewer/issues/11), fixed now)
* No forward search and inverse search.
* When the pdf is open in IJ, it will always refresh this pdf and ignore whatever viewer you have selected. (Bug or feature?)

**Note**
The IntelliJ PDF Viewer plugin is still in early stage, which means that it might have bugs and lacks some features.

#### How to test this pull request

* Select the `Built-in PDF Viewer` option for the pdf viewer.
* Compile, the pdf should be opened inside the editor on the right side.
* Close the pdf in IJ. 
* Select a different pdf viewer and compile (to see that that still works).
* Disable the IntelliJ PDF Viewer plugin.
* Check that it is not available in the list of pdf viewers.

#### Wiki
